app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php

```diff
@@ -84,6 +84,8 @@ public function execute(string $resetPasswordToken):CustomerInterface
         }
 
         //Unique customer found.
-        return $found->getItems()[0];
+        $items = $found->getItems();
+
+        return reset($items);
```

Comment:
If the `$items` is an empty array `reset` function would return `false` and PHP would throw a fatal error, I guess it worth checking array length and throw an exception on such case, what do you think?